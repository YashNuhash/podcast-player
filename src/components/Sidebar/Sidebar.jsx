import React, { useState } from 'react'
import { NavLink, Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import { selectPlayingPodcastId } from '../../redux/playing/playingSelectors'
import { ReactComponent as HomeIcon } from '../../assets/home.svg'
import { ReactComponent as SearchIcon } from '../../assets/search.svg'
import { ReactComponent as PodcastIcon } from '../../assets/podcast.svg'
import styles from './Sidebar.module.scss'

export const Sidebar = ({ playingPodcastId }) => {
    const [isSidebarVisible, setIsSidebarVisible] = useState(false)

    return (
        <>
            <span
                aria-label='Toggle menu'
                onClick={() => setIsSidebarVisible(!isSidebarVisible)}
                className={styles.Toggle}
            >
                <span
                    className={`${styles.ToggleIcon} ${
                        isSidebarVisible ? styles.ToggleIconWhenVisible : ''
                    }`}
                ></span>
            </span>

            <div
                className={`${styles.Sidebar} ${
                    isSidebarVisible ? styles.SidebarMobileVisible : ''
                }`}
            >
                <h1 className={styles.Logo}>
                    <Link to='/' className={styles.LogoLink}>
                        Podcast
                    </Link>
                </h1>

                <nav className={styles.Nav}>
                    <h4 className={styles.Title}>PODCAST</h4>
                    <ul>
                        <li className={styles.ListItem}>
                            <NavLink
                                exact
                                to='/'
                                className={styles.Link}
                                activeClassName={styles.ActiveLink}
                            >
                                <HomeIcon className={styles.LinkIcon} />
                                Home
                            </NavLink>
                        </li>

                        <li className={styles.ListItem}>
                            <NavLink
                                to='/discover'
                                className={styles.Link}
                                activeClassName={styles.ActiveLink}
                            >
                                <SearchIcon className={styles.LinkIcon} />
                                Discover
                            </NavLink>
                        </li>

                        <li className={styles.ListItem}>
                            <NavLink
                                to={`/podcast/${playingPodcastId}`}
                                className={styles.Link}
                                activeClassName={
                                    playingPodcastId ? styles.ActiveLink : ''
                                }
                            >
                                <PodcastIcon className={styles.LinkIcon} />
                                Now Playing
                            </NavLink>
                        </li>
                    </ul>
                </nav>
            </div>
        </>
    )
}

const mapStateToProps = createStructuredSelector({
    playingPodcastId: selectPlayingPodcastId,
})

export default connect(mapStateToProps)(Sidebar)
