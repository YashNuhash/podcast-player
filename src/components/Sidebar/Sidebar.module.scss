@import '../../utils';

.Sidebar {
    position: fixed;
    width: 20rem;
}

.Nav {
    height: 100%;
}

.Logo {
    font-size: 2.8rem;
    padding-left: 2rem;
    margin-bottom: 10rem;
}

.LogoLink {
    color: $primary;
    text-decoration: none;
}

.Title {
    color: $light-text;
    padding: 0 2rem;
    font-size: 1.5rem;
    margin-bottom: 2rem;
}

.ListItem {
    list-style: none;
    display: flex;
    margin: 1rem 0;
}

.Link {
    display: inline-flex;
    align-items: center;
    width: 100%;
    color: $text;
    text-decoration: none;
    font-weight: 600;
    font-size: 1.4rem;
    padding: 1rem 2rem;
}

.LinkIcon {
    fill: $text;
    width: 1.6rem;
    height: 1.6rem;
    margin-right: 1rem;
}

.ActiveLink {
    background-color: $text;
    color: $offset;
    border-radius: 100px;

    .LinkIcon {
        fill: $offset;
    }
}
