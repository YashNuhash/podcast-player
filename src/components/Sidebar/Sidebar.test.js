import React from 'react'
import { shallow } from 'enzyme'
import { Sidebar } from './Sidebar'

describe('Sidebar', () => {
    it('should render component', () => {
        expect(shallow(<Sidebar />)).toMatchSnapshot()
    })
    it('should render component when playing podcast', () => {
        expect(shallow(<Sidebar playingPodcastId='1' />)).toMatchSnapshot()
    })
    it('should render component when sidebar is visible', () => {
        const wrapper = shallow(<Sidebar />)
        wrapper
            .find('span')
            .first()
            .simulate('click')
        expect(wrapper).toMatchSnapshot()
    })
})
