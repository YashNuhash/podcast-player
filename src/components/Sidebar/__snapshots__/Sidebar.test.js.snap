// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render component 1`] = `
<Fragment>
  <span
    aria-label="Toggle menu"
    className="Toggle"
    onClick={[Function]}
  >
    <span
      className="ToggleIcon "
    />
  </span>
  <div
    className="Sidebar "
  >
    <h1
      className="Logo"
    >
      <Link
        className="LogoLink"
        to="/"
      >
        Podcast
      </Link>
    </h1>
    <nav
      className="Nav"
    >
      <h4
        className="Title"
      >
        PODCAST
      </h4>
      <ul>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            exact={true}
            to="/"
          >
            <ForwardRef(SvgHome)
              className="LinkIcon"
            />
            Home
          </NavLink>
        </li>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            to="/discover"
          >
            <ForwardRef(SvgSearch)
              className="LinkIcon"
            />
            Discover
          </NavLink>
        </li>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName=""
            className="Link"
            to="/podcast/undefined"
          >
            <ForwardRef(SvgPodcast)
              className="LinkIcon"
            />
            Now Playing
          </NavLink>
        </li>
      </ul>
    </nav>
  </div>
</Fragment>
`;

exports[`Sidebar should render component when playing podcast 1`] = `
<Fragment>
  <span
    aria-label="Toggle menu"
    className="Toggle"
    onClick={[Function]}
  >
    <span
      className="ToggleIcon "
    />
  </span>
  <div
    className="Sidebar "
  >
    <h1
      className="Logo"
    >
      <Link
        className="LogoLink"
        to="/"
      >
        Podcast
      </Link>
    </h1>
    <nav
      className="Nav"
    >
      <h4
        className="Title"
      >
        PODCAST
      </h4>
      <ul>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            exact={true}
            to="/"
          >
            <ForwardRef(SvgHome)
              className="LinkIcon"
            />
            Home
          </NavLink>
        </li>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            to="/discover"
          >
            <ForwardRef(SvgSearch)
              className="LinkIcon"
            />
            Discover
          </NavLink>
        </li>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            to="/podcast/1"
          >
            <ForwardRef(SvgPodcast)
              className="LinkIcon"
            />
            Now Playing
          </NavLink>
        </li>
      </ul>
    </nav>
  </div>
</Fragment>
`;

exports[`Sidebar should render component when sidebar is visible 1`] = `
<Fragment>
  <span
    aria-label="Toggle menu"
    className="Toggle"
    onClick={[Function]}
  >
    <span
      className="ToggleIcon ToggleIconWhenVisible"
    />
  </span>
  <div
    className="Sidebar SidebarMobileVisible"
  >
    <h1
      className="Logo"
    >
      <Link
        className="LogoLink"
        to="/"
      >
        Podcast
      </Link>
    </h1>
    <nav
      className="Nav"
    >
      <h4
        className="Title"
      >
        PODCAST
      </h4>
      <ul>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            exact={true}
            to="/"
          >
            <ForwardRef(SvgHome)
              className="LinkIcon"
            />
            Home
          </NavLink>
        </li>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName="ActiveLink"
            className="Link"
            to="/discover"
          >
            <ForwardRef(SvgSearch)
              className="LinkIcon"
            />
            Discover
          </NavLink>
        </li>
        <li
          className="ListItem"
        >
          <NavLink
            activeClassName=""
            className="Link"
            to="/podcast/undefined"
          >
            <ForwardRef(SvgPodcast)
              className="LinkIcon"
            />
            Now Playing
          </NavLink>
        </li>
      </ul>
    </nav>
  </div>
</Fragment>
`;
